### PR TITLE
[CI] Change `Regression Test new micro benchmark` job to diff against target branch

### PR DESCRIPTION
**Pull request by Tishj**
This PR aims to fix CI of PRs made to `feature`:

```
Run git diff --name-only origin/main | (grep "benchmark/micro/*" || true) > .github/regression/updated_micro_benchmarks.csv
Detected the following modified benchmarks. Running a regression test
benchmark/micro/csv/ignore_errors.benchmark
benchmark/micro/csv/multiple_small_files.benchmark
benchmark/micro/csv/time_type.benchmark
benchmark/micro/result_collection/batched_stream_query.benchmark
```

```
benchmark/micro/csv/ignore_errors.benchmark
Old timing: 2.205773
New timing: Failed to run benchmark benchmark/micro/csv/ignore_errors.benchmark

benchmark/micro/csv/multiple_small_files.benchmark
Old timing: 0.244721
New timing: Failed to run benchmark benchmark/micro/csv/multiple_small_files.benchmark

benchmark/micro/csv/time_type.benchmark
Old timing: 0.921153
New timing: Failed to run benchmark benchmark/micro/csv/time_type.benchmark

benchmark/micro/result_collection/batched_stream_query.benchmark
Old timing: 7.897132
New timing: Failed to run benchmark benchmark/micro/result_collection/batched_stream_query.benchmark
```

By only running this job if the event is a PR, and diffing against the target branch instead of hardcoding this to `main`

### CODE REVIEW

**Comment by Tishj** [comment]
@Tmonster tagging you since you made this change originally, wanted to know if this makes sense / does not go against the intended behavior for this job

**Review by carlopi — CHANGES_REQUESTED** [review]
Looking at this with fresh eyes: could we not use the BASE_BRANCH env variable and setting that correctly in case of pull requests to use the right value?

E

**Review by carlopi — CHANGES_REQUESTED** [review]
I think the proper way forward would be:

using instead of  `--benchmarks=.github/regression/imdb.csv` the file present in the base_ref, so `--benchmarks=duckdb/.github/regression/imdb.csv` (I think, to be double checked), this can be done everywhere for executing the original tests instead of the new ones.

Possibly, some checks on newly introduced benchmarks, at least to test the that they are working. But that can also be left out for now.

File: .github/workflows/Regression.yml [hunk 1]
```diff
@@ -130,10 +130,12 @@ jobs:
         python scripts/regression_test_runner.py --old=duckdb/build/release/benchmark/benchmark_runner --new=build/release/benchmark/benchmark_runner --benchmarks=.github/regression/csv.csv --verbose --threads=2
 
     - name: Regression Test new micro benchmark
-      if: always()
+      if: github.event_name == 'pull_request'
```
Comment (Tmonster) [inline .github/workflows/Regression.yml:133]: I wouldn't do this on pull request. It means this only triggers when the pull request is opened, but not when a user pushes updates to their branch

**Comment by Tmonster** [comment]
I think just pointing to the `target` branch is fine. And I think you need to add the remote ref too, otherwise there can be another failure

**Comment by Tishj** [comment]
> I think just pointing to the `target` branch is fine. And I think you need to add the remote ref too, otherwise there can be another failure

I don't follow, a push doesn't have a target branch, and you said we would need to do on pushes as well?
The remote ref would be required if the PR was being made to a fork right? Yea I agree that could fail

Judging from the hack we have in place for `BASE_BRANCH` I think it's not possible to get the target branch for a push event

**Comment by Tishj** [comment]
Although.. digging into the docs, I think an event can have an associated pull request?

**Comment by Tmonster** [comment]
Synchronize looks like it could work! I thought I looked into something like that, but I couldn't find the documentation. Hopefully it's still supported

**Comment by Tishj** [comment]
@carlopi I poked around a bit in the pull request payload data, I wanted to get your opinion on this
This should make it nicer to run CI on forks, the idea is to call this function inside `if:` of steps

```js
exports.doNotRunOnDraft = function(github) {
	const is_draft = isDraft(github);
	const is_opened = checkPullRequestAction(github, 'opened');
	const is_reopened = checkPullRequestAction(github, 'reopened');
	const is_synchronize = checkPullRequestAction(github, 'synchronize')
	const is_ready_for_review = checkPullRequestAction(github, 'ready_for_review');
	if (github.event.pull_request.base.repo.html_url === 'https://github.com/duckdb/duckdb') {
		if (is_synchronize) {
			// synchronize (push to open PR) should never trigger workflows if PR is to the upstream repo
			return false;
		}
		// PR is made to upstream repo
		if (is_draft) {
			// draft PRs should never run CI when they're made to the upstream repo
			return false;
		}
		if (is_opened) {
			return true;
		}
		if (is_reopened) {
			return true;
		}
		if (is_ready_for_review) {
			return true;
		}
		return false;
	} else {
		// PR is made to a fork, should always run
		return true;
	}
};
```

Some issues this *should* solve:
PR opened to upstream as draft still runs CI currently, which it shouldnt
forks use the same strategy as upstream, canceling all CI and auto drafting PRs that are made to forks

**Review by carlopi — CHANGES_REQUESTED** [review]
I think this can be experimented on, change here is pretty innocuous and I don't see what could go wrong since it's limited to a specific workflow.
Idea is very neat, thanks for investigating and finding this.

I would say medium term it could be nice to have the same solution across workflows, and possibly revisit / improve on current infrastructure to get to a more solid place across the board.

On PRs specific, I would probably move helper.js down a subfolder level and consider changing the file name to be more descriptive.

**Comment by Tishj** [comment]
With this change the Regression workflow doesn't run

This was a good experiment I think, but I'm shelfing it for now

**Pull request closed without merging**